kolibri/plugins/coach/assets/src/vue/groups-page/index.vue

```diff
@@ -1,71 +1,159 @@
 <template>
 
-  <div>
-    <h1>{{ $tr('groups') }} - {{ className }}</h1>
-    <icon-button
-      :text="$tr('newGroup')"
+  <div class="wrapper">
+    <h1>{{ className }} - {{ $tr('groups') }}</h1>
+    <icon-button :text="$tr('newGroup')"
       :primary="true"
-      @click="openCreateGroupModal"
-    >
-      <mat-svg category="content" name="add"/>
+      @click="openCreateGroupModal">
+      <mat-svg category="content"
+        name="add" />
     </icon-button>
 
-    <create-group-modal
-      v-if="showCreateGroupModal"
+    <create-group-modal v-if="showCreateGroupModal"
       :className="className"
-      :classId="classId"/>
+      :classId="classId" />
 
-    <div v-for="group in groups">
-      <h2>{{ group.name }}</h2>
-      <!--{{ $tr('numLearners', {count: group.users.length}) }} -->
-    </div>
+    <rename-group-modal v-if="showRenameGroupModal"
+      :className="className"
+      :classId="classId"
+      :groupName="selectedGroup.name"
+      :groupId="selectedGroup.id" />
+
+    <delete-group-modal v-if="showDeleteGroupModal"
+      :className="className"
+      :classId="classId"
+      :groupName="selectedGroup.name"
+      :groupId="selectedGroup.id" />
+
+    <move-learners-modal v-if="showMoveLearnersModal"
+      :className="className"
+      :classId="classId"
+      :groupName="selectedGroup.name"
+      :groupId="selectedGroup.id"
+      :groups="groups"
+      :usersToMove="usersToMove"
+      :isUngrouped="isUngrouped" />
+
+    <group-section v-for="group in groups"
+      :group="group"
+      :className="className"
+      :classId="classId"
+      @rename="openRenameGroupModal"
+      @delete="openDeleteGroupModal"
+      @move="openMoveLearnersModal" />
+
+    <group-section :group="ungroupedUsersObject"
+      :className="className"
+      :classId="classId"
+      :isUngrouped="true"
+      @move="openMoveLearnersModal" />
   </div>
 
 </template>
 
 
 <script>
 
-  const actions = require('../../actions');
+  const groupActions = require('../../group-actions');
   const constants = require('../../state/constants');
-
+  const differenceWith = require('lodash.differencewith');
```

Comment:
I remember hearing that we're not including lodash in the frontend... but I might have misunderstood. @rtibbles ?